### ISSUE TITLE
Package Replication Status Updater throws exceptions when packages has require admin flag

### ISSUE THREAD

**Issue by kwin** [Adobe-Consulting-Services/acs-aem-commons#1478]
## Required Information

* [ ] AEM Version, including Service Packs, Cumulative Fix Packs, etc: 6.3.2.1
* [ ] ACS AEM Commons Version: 3.17.4
* [ ] Reproducible on Latest? yes

### Expected Behavior

The Package Replication Status Updater should not fail in case you install packages it has the require-admin flag set

### Actual Behavior

The status updater fails with the following error
```
05.09.2018 11:33:21.673 *WARN* [sling-threadpool-ff2c203d-3904-4666-9dc0-32105c3275c5-(apache-sling-job-thread-pool)-6-<main queue>(acs-commons/replication/package)] com.adobe.acs.commons.replication.status.impl.JcrPackageReplicationStatusEventHandler Job failed with error 'Could not update replication metadata' in attempt '2', retry later.
javax.jcr.RepositoryException: Could not retrieve the Packages contents for package '<my package>'
	at com.adobe.acs.commons.replication.status.impl.JcrPackageReplicationStatusEventHandler.setReplicationStatus(JcrPackageReplicationStatusEventHandler.java:311)
	at com.adobe.acs.commons.replication.status.impl.JcrPackageReplicationStatusEventHandler.process(JcrPackageReplicationStatusEventHandler.java:251)
	at org.apache.sling.event.impl.jobs.JobConsumerManager$JobConsumerWrapper.process(JobConsumerManager.java:500)
	at org.apache.sling.event.impl.jobs.queues.JobQueueImpl.startJob(JobQueueImpl.java:291)
	at org.apache.sling.event.impl.jobs.queues.JobQueueImpl.access$100(JobQueueImpl.java:58)
	at org.apache.sling.event.impl.jobs.queues.JobQueueImpl$1.run(JobQueueImpl.java:227)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.jackrabbit.vault.packaging.PackageException: Package extraction requires admin session (userid not allowed).
	at org.apache.jackrabbit.vault.packaging.impl.ZipVaultPackage.prepareExtract(ZipVaultPackage.java:188)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:390)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:357)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:525)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:357)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:525)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:357)
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:343)
	at com.adobe.acs.commons.packaging.impl.PackageHelperImpl.getContents(PackageHelperImpl.java:318)
	at com.adobe.acs.commons.replication.status.impl.JcrPackageReplicationStatusEventHandler.setReplicationStatus(JcrPackageReplicationStatusEventHandler.java:304)
	... 8 common frames omitted
```

### Steps to Reproduce

Just enable the replication status updater (by setting a configuration) and install a package with the require-admin flag set to true.

**Comment by kwin** [comment]
This was always an issue,  but not noticed due to the incorrect exception handling prior to https://github.com/Adobe-Consulting-Services/acs-aem-commons/commit/7048b71b63590c9175faad15966d1e521a159140

**Comment by kwin** [comment]
I asked at https://www.mail-archive.com/dev@jackrabbit.apache.org/msg42382.html whether dry-run installing packages with hooks should really require admin rights. This doesn't feel right to me.

**Comment by kwin** [comment]
One related problem is that potentially multiple packages are treated in the same job. If one package fails the whole job is failed (i.e. other packages are not processed). I think it therefore makes sense to only process one package per job. I will create a PR for that.

**Comment by davidjgonzalez** [comment]
Fix for this issues released in 3.18.2

**Comment by kwin** [comment]
@davidjgonzalez Only one specific issue was fixed in 3.18.2 namely the issue mentioned in https://github.com/Adobe-Consulting-Services/acs-aem-commons/issues/1478#issuecomment-419340019. There is still the issue that the system users does not belong to the admin user group, i.e. packages having the "admin" flag will throw an exception and their replication status is not properly tracked. Would you mind reopening this issue?
